.github/workflows/test.yml

```diff
@@ -104,10 +104,41 @@ jobs:
         run: |
           rm -f changed_files.txt
 
-  yarn-test:
+  rust-only:
+    outputs:
+      only_rust: ${{ steps.check-rust-only.outputs.only_rust }}
     runs-on: ubuntu-latest
-    needs: [yarn-install]
+    env:
+      PR_BASE: ${{ github.base_ref || 'HEAD^' }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
```

Comment:
When this is in the merge queue, PR-based context values are not available.
- The event type is `merge_group`
- `github.base_ref` and `github.head_ref` are not set
- You would use `github.event.merge_group.base_sha` and `github.event.merge_group.head_sha`
- Or `github.event.merge_group.base_ref` and `github.event.merge_group.head_ref`

See https://docs.github.com/en/webhooks/webhook-events-and-payloads#merge_group

Reply:
yeah but then it's fine to stick with `github.sha` when on branch/queue right, because we want to check out the commit itself not whatever is at the head of the queue?

Reply:
ah your comment here was about the other line not this checkout itself